test/Starcounter.ErrorCodes.Tests/ExceptionFactoryTests.cs

```diff
@@ -0,0 +1,96 @@
+namespace Starcounter.ErrorCodes.Tests
+{
+    using System;
+    using System.IO;
+    using Xunit;
+
+    public class ExceptionFactoryTests
+    {
+        public class TheCreateExceptionClass
+        {
+            [Fact]
+            public void ThrowsExceptionWhenMessageFactoryIsNull()
+            {
+                //Given
+                var factory = new ExceptionFactory();
+
+                //When
+                Action createException = () => factory.CreateException(0, null, null, null, null);
```

Comment:
This is not guaranteed to assert what the test says. You need to provide valid arguments for all **but** the message factory. Otherwise, you can't know what triggered the exception.

```suggestion
                Action createException = () => factory.CreateException(
                  0, 
                  new Exception("Dummy"), 
                  "dummy", 
                  null, 
                  new [] {string.Empty)};
```

Reply:
Good point! Fixed